onnxruntime/core/optimizer/qdq_transformer/selectors_actions/qdq_selector_action_transformer.cc

```diff
@@ -105,8 +105,8 @@ void UnaryOpQDQRules(SelectorActionRegistry& qdq_selector_action_registry) {
   std::unique_ptr<Action> action = std::make_unique<QDQ::UnaryReplaceWithQLinear>(kMSDomain);
 
 #if !defined(ORT_MINIMAL_BUILD)
-  // TODO: Enable 16-bit types in selector when unary QLinear* ops support 16-bit.
-  std::unique_ptr<NodeSelector> selector = std::make_unique<QDQ::UnarySelector>();
+  std::vector<const char*> providers = {kCpuExecutionProvider};
+  std::unique_ptr<NodeSelector> selector = std::make_unique<QDQ::UnarySelector>(providers);
```

Comment:
Should this list also contain the `kDmlExecutionProvider`, if we want to enable this rule for DML EP?
Or is it intentional? Meaning the rules which don't have any `providers` list will be applicable to both CPU EP and DML EP. And the rules which has `provider` list that will be applicable only to CPU ep.
Adding @jeffbloo as well as he might have more context since he added it.

Reply:
Thanks for checking.  This is intentional based on current operator support of the DML EP.